src/_common/report/form/form.ts

```diff
@@ -1,22 +1,309 @@
 import { Component, Prop } from 'vue-property-decorator';
-
-import { FormOnSubmit, BaseForm } from '../../form-vue/form.service';
+import { arrayRemove } from '../../../utils/array';
 import { Api } from '../../api/api.service';
+import { FiresidePost } from '../../fireside/post/post-model';
+import AppFormControlTextarea from '../../form-vue/control/textarea/textarea.vue';
+import { BaseForm, FormOnSubmit } from '../../form-vue/form.service';
+import { Game } from '../../game/game.model';
+
+interface FormModel {
+	reason: string;
+	context: string[];
+	description: string;
+	source: string;
+}
 
-@Component({})
-export default class AppReportForm extends BaseForm<any> implements FormOnSubmit {
+@Component({
+	components: {
+		AppFormControlTextarea,
+	},
+})
+export default class AppReportForm extends BaseForm<FormModel> implements FormOnSubmit {
 	@Prop(String) type!: string;
 	@Prop(Object) resource!: any;
 
 	warnOnDiscard = false;
 
+	// Default values, get overwritten from backend.
+	maxLengthDescription = 200;
+	maxLengthSource = 255;
+
+	get valid() {
+		if (!this._isValid()) {
+			return false;
+		}
+
+		// Check that if we have contexts to choose from, at least one of them is selected.
+		if (!this.formModel.reason) {
+			return true;
+		}
+		const reason = this.reasons.find(i => i.radioValue === this.formModel.reason);
+		if (!reason) {
+			return true;
+		}
```

Comment:
we should probably be throwing here, yeah? theres no such thing as a non existing reason object.